SinGAN/training.py

```diff
@@ -172,7 +172,7 @@ def train_single_scale(netD,netG,reals,Gs,Zs,in_s,NoiseAmp,opt,centers=None):
 
         for j in range(opt.Gsteps):
             netG.zero_grad()
-            output = netD(fake)
+            output = netD(fake.detach())

```

Comment:
Hi,

I am wondering if you change to fake.detach(), whether this means you exclude the netG out of the computation graph. And we don't update netG anymore? Thanks.

Reply:
Hi Alex Wei!
I agree with gap370's opinion, if use fake.detach(), netG will not update anymore.

Reply:
Hi Alex Wei!
I agree with gap370's opinion, if use fake.detach(), netG will not update anymore.
After Pytorch 1.5.0, the reason for the error may be because

Reply:
Hi Alex Wei!
I agree with gap370's opinion, if use fake.detach(), netG will not update anymore.
After Pytorch 1.5.0, the reason for the error may be because

Reply:
Hi Alex Wei!
I agree with gap370's opinion, if use fake.detach(), netG will not update anymore.
I think maybe we need to generate a new fake to update netG, and errG don't use retain_graph=true
`
for j in range(opt.Gsteps):                                                                                                           
	netG.zero_grad()     
	# create a new fake here                                                                                                              
	fake = netG(noise.detach(),prev)                                                                                                 
	output = netD(fake)                                                                                                                   
	#D_fake_map = output.detach()                                                                                                 
	errG = -output.mean()                                                                                                               
	errG.backward() 
`